### ISSUE TITLE
Better color flags for terminal output

### ISSUE THREAD

**Issue by UnlimitedCookies** [yt-dlp/yt-dlp#3643]
### Checklist

- [X] I'm reporting a feature request
- [X] I've looked through the [README](https://github.com/yt-dlp/yt-dlp#readme)
- [X] I've verified that I'm running yt-dlp version **2022.04.08** ([update instructions](https://github.com/yt-dlp/yt-dlp#update)) or later (specify commit)
- [X] I've searched the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues including closed ones. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)

### Description

Currently there only exists a `--no-colors` option. This only allows disabling color output, but there is no option to always enable color output. This is useful in my case, because I want to pipe it into grep, while also showing its output using tee.
That's why I propose the standardized way many applications (e.g. ls and grep) in the unix world already use:
```
--color=auto (default)
--color=always
--color=never
```

**Comment by sulyi** [comment]
Default behaviour is already `--color=auto`, and `--no-color` is already works as `--color=never`.
Unless [`io.IOBase.isatty`](https://docs.python.org/3/library/io.html#io.IOBase.isatty) fails somehow.

**Comment by pukkandan** [comment]
@sulyi What they want is a `color=always` option

**Comment by UnlimitedCookies** [comment]
> @sulyi What they want is a `color=always` option so that color codes can be piped to grep

Exactly. At the same time I am proposing to use the same syntax, as other Unix tools already do.

**Comment by sulyi** [comment]
> they want is a `color=always`

Yes.

 > same syntax, as other Unix tools

There is no exact standard, tho, e.g. my `grep` version has only `--color` option and my `ls` has both `--color` and `--colour`. If I understand correctly, using `auto`, `always`, and `never` as possible values was requested.

Also there's the issue with stderr: I've decided it will always detect stream type, as is the case in many cases and there's existing workaround to handle it, (see: https://stackoverflow.com/questions/27397865/how-to-write-stdout-to-file-with-colors). This behaviour might change after implementing #2668. I'm not sure if python is able to comply, tho:

> Unless io.IOBase.isatty fails somehow.

And there were different platforms, but I try not to think about them for now.

PR is coming soon.

**Comment by UnlimitedCookies** [comment]
> > 
> 
> There is no exact standard, tho, e.g. my `grep` version has only `--color` option and my `ls` has both `--color` and `--colour`. If I understand correctly, using `auto`, `always`, and `never` as possible values was requested.

Yeah, I know it's not standardized in any way, but many applications recognize it already, and it's fairly widespread.

>>Unless io.IOBase.isatty fails somehow.
>And there were different platforms, but I try not to think about them for now.

It's not a tty, because it first goes through tee.

**Comment by Grub4K** [comment]
> Exactly. At the same time I am proposing to use the same syntax, as other Unix tools already do.

I would propose something more granular:
- `--color always`: Apply to all outputs
- `--color screen:always --color stdout:auto --color stderr:never`: Granular control
- `--color sceen:terminal`: Would enable terminal sequences but not color sequences

`auto` would then check `output.isatty()`, then `environ['TERM'] != 'dumb'`.

This keeps compat with the common posix way while allowing fined control. All of these can be implemented without output rework.

**Comment by pukkandan** [comment]
> --color sceen:terminal: Would enable terminal sequences but not color sequences

I don't get the use case of this. Rest sounds good.

**Comment by Grub4K** [comment]
Some people don't want colored output in their terminal but not suffer from multiline progress and console title not working.

**Comment by pukkandan** [comment]
`stdout:nocolor`?
